packages/connect/src/device/DeviceCommands.ts

```diff
@@ -348,43 +319,77 @@ export class DeviceCommands {
-        try {
-            assertType(response, resType);
-        } catch (error) {
-            // handle possible race condition
-            // Bridge may have some unread message in buffer, read it
-            const abortController = new AbortController();
-            const timeout = setTimeout(() => {
-                abortController.abort();
-            }, 500);
-
-            await this.transport
-                .receive({
-                    session: this.transportSession,
-                    protocol: this.device.protocol,
-                    signal: abortController.signal,
-                })
-                .finally(() => {
-                    clearTimeout(timeout);
-                });
-            // throw error anyway, next call should be resolved properly
-            throw error;
+
+        if ('isTransportError' in response) {
+            throw new Error(response.error, { cause: 'transport-error' });
         }
 
-        return response;
+        const splitResTypes = Array.isArray(resType) ? resType : resType.split('|');
+        if (splitResTypes.includes(response.type)) {
+            return response;
+        }
+        // handle possible race condition
+        // Bridge may have some unread message in buffer, read it
+        const abortController = new AbortController();
+        const timeout = setTimeout(() => {
+            abortController.abort();
+        }, 500);
+
+        await this.transport
+            .receive({
+                session: this.transportSession,
+                protocol: this.device.protocol,
+                signal: abortController.signal,
+            })
+            .finally(() => {
+                clearTimeout(timeout);
+            });
+
+        // throw error anyway, next call should be resolved properly
+
+        throw ERRORS.TypedError(
+            'Runtime',
+            `assertType: Response of unexpected type: ${response.type}. Should be ${resType}`,
+        );
     }
 
-    async _commonCall(type: MessageKey, msg?: DefaultPayloadMessage['message']) {
+    async _commonCall(
+        type: MessageKey,
+        msg?: DefaultPayloadMessage['message'],
+    ): Promise<
+        DefaultPayloadMessage | (ErrorGeneric<ReadWriteError> & { isTransportError: true })
+    > {
         if (this.disposed) {
             throw ERRORS.TypedError('Runtime', 'typedCall: DeviceCommands already disposed');
         }
-        const resp = await this.call(type, msg);
+        const res = await this.call(type, msg);
+
+        if (!res.success) {
+            logger.warn(
+                'Received error',
+                res.error,
+                // res.message is not propagated to higher levels, only logged here. webusb/node-bridge may return message with additional information
+                res.message,
+            );
+
+            return {
+                ...res,
+                isTransportError: true,
+            };
```

Comment:
Given that `_commonCall` is not throwing any more in this case, it will change the logic below in `_filterCommonTypes`, e.g. `this._commonCall('PinMatrixAck', { pin }).then(response => ...)`, right? Do we mind?